src/app/auth.guard.ts

```diff
@@ -33,9 +33,18 @@ export class AuthGuard implements CanActivate {
         if (result) {
           return true;  // Allow access
         }
+
+        
         // If no bypass in effect and user role is not in role restriction, then no access
-        if (roles.indexOf(JSON.parse(this.auth.getProfile()).app_metadata.roles) == -1)
-        {
+        
+        var access: boolean = 
```

Comment:
It was made clear however that `roles.indexOf` was intended for the conditional check.  Here, each user role in `app_metadata.roles` is matched against any listed router `roles` for the given URL endpoint, as listed in `app-routing.module.ts`. A `map` function is used here to evaluate the index of each role in `app_metadata.roles`.